sisl/grid.py

```diff
@@ -139,7 +139,30 @@ def fill(self, val):
         """
         self.grid.fill(val)
 
-    def interp(self, shape, method='linear', **kwargs):
+    def interp(self, shape, order=1, **kwargs):
+        """ 
+        Interpolate grid values to a new grid.
+
+        It uses the `zoom` method in `scipy.ndimage`, which creates a finer or
+        # more spaced grid using spline interpolation
+
+        Parameters
+        ----------
+        shape : int, array_like
+            the new shape of the grid
+        order : int 0-5, optional
+            the order of the spline interpolation. 
+            1 means linear, 2 quadratic, etc...
+        **kwargs : dict
+            optional arguments passed to the interpolation algorithm
+            The interpolation routine is `scipy.ndimage.zoom`
+        """
+        # Calculate the zoom_factors
+        zoom_factors = np.array(shape) / self.shape
+
+        return self.apply('zoom', zoom_factors, mode='wrap', order=order, **kwargs)
+
+    def old_interp(self, shape, method='linear', **kwargs):
```

Comment:
Just delete the old one, to ensure we don't break old code we could put in a check `kwargs.get("method", None)` and check for "linear". Then set corresponding values.